### PR TITLE
HPCC-28793 Refactor docker base build tagging

### PR DESCRIPTION
**Pull request by GordonSmith**
New format:  hpccsystems/platform-build-base-centos-7:2022.11.14-rc4

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by GordonSmith** [comment]
@ghalliday - rebased

File: .github/workflows/build-assets.yml [hunk 1]
```diff
@@ -149,11 +157,11 @@ jobs:
         run: |
           sudo rm -f ./build/CMakeCache.txt
           sudo rm -rf ./build/CMakeFiles
-          docker_label=${{ secrets.DOCKER_USERNAME }}/build-${{ matrix.os }}:${{ steps.vars.outputs.branch_name }}
+          docker_label=${{ secrets.DOCKER_USERNAME }}/build-${{ matrix.os }}:${{ steps.vars.outputs.community_tag }}
```
Comment (jakesmith) [inline .github/workflows/build-assets.yml:160]: question: this is the LN cmake step, should this be internal_tag?

File: dockerfiles/vcpkg/ubuntu-22.04.dockerfile [hunk 2]
```diff
@@ -1,6 +1,5 @@
 ARG VCPKG_REF=latest
-ARG DOCKER_NAMESPACE=hpccbuilds
-FROM ${DOCKER_NAMESPACE}/vcpkg-ubuntu-22.04:${VCPKG_REF}
+FROM hpccsystems/platform-build-base-ubuntu-22.04:$VCPKG_REF
```
Comment (jakesmith) [inline dockerfiles/vcpkg/ubuntu-22.04.dockerfile:2]: question: is "hpccbuilds" no longer used, everything to/from "hpccsystems" ?

**Review by jakesmith — COMMENTED** [review]
@GordonSmith: I think it can be approved, 1 real question re. community vs internal docker label

File: dockerfiles/vcpkg/ubuntu-22.04.dockerfile (hunk 2 above)
Comment (GordonSmith) [reply dockerfiles/vcpkg/ubuntu-22.04.dockerfile:2]: It is for some of the caches

File: .github/workflows/build-assets.yml (hunk 1 above)
Comment (GordonSmith) [reply .github/workflows/build-assets.yml:160]: No - its using the build docker image from community to build the LN sources.

File: dockerfiles/vcpkg/build.sh [hunk 3]
```diff
@@ -22,7 +22,7 @@ echo "VCPKG_REF: $VCPKG_REF"
 echo "DOCKER_USERNAME: $DOCKER_USERNAME"
 echo "DOCKER_PASSWORD: $DOCKER_PASSWORD"
 
-docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD
+# docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD
```
[the remark this answers is not in the document]
Comment (GordonSmith) [reply dockerfiles/vcpkg/build.sh:25]: Just a dev/testing tool - will probably get removed with:  https://track.hpccsystems.com/browse/HPCC-28797

File: .github/workflows/build-assets.yml (hunk 1 above)
Comment (jakesmith) [reply .github/workflows/build-assets.yml:160]: ok

**Review by ghalliday — APPROVED** [review]

**Review by jakesmith — APPROVED** [review]
Can't see any issues.

**Pull request merged**